### DISCUSSION TITLE
Nextcloud AIO Login password changes after backup : normal behaviour ?

### DISCUSSION THREAD

**Discussion by tdeseez** [nextcloud/all-in-one#2832] [Questions] [answered]
Hi,

I just tested the backup of my AIO installation with built-in Borg container and it went like a charm.

But AIO login password is not restored as it was from last installation. Since this is a backup, i thought everything would be reset as before ?

This may be confusing : after a backup you have to remember another password. Isn't this the best way to eventually lose it ? 

Thanks for your answer.

regards/
thierry

**Comment by szaimen** [comment, marked as answer]
Hi, interesting discussion. Actually the main idea why this was implemented was that it would be confusing if one would change the password via the aio interface, restore the last backup state and then having suddently the old password again which one might have overwritten already. Here it comes with the sideffect that it also keeps the newly generated password on an instance restore. However from my perspective it makes sense from a security standpoint to keep this behaviour as keeping the newly generated one is good to change it once every few years or months. Additionally, if you are lost which one it was, you can pretty easily retrieve it by so: https://github.com/nextcloud/all-in-one/discussions/1786

**Comment by tdeseez** [comment]
Thanks for answering.
If you think it is a security concern, maybe it's better to keep it this way. 

Just got this idea : give the user an option about the password during the backup process (keep the old one or go for the new one) ? 

Anyway, I didn't know this was that easy to retrieve the AIO password, thanks for the tip.

By the way, is there a similar tip to recover the Borg password ?

Thanks again/
thierry

**Reply by szaimen** [reply]
> Anyway, I didn't know this was that easy to retrieve the AIO password, thanks for the tip.

You are welcome!

> By the way, is there a similar tip to recover the Borg password ?

It is also stored in the mentioned file or do you e.g. want to retreive it on a new instance after oruning all files from an old one?

**Comment by tdeseez** [comment]
Hi
Is there a difference between "borg_restore_password" and "BORGBACKUP_PASSWORD" ?

Cause when i run `sudo docker exec nextcloud-aio-mastercontainer grep password /mnt/docker-aio-config/data/configuration.json`, the "borg_restore_password" key returns an empty string, while the "password" key is ok.

if i run `sudo docker exec nextcloud-aio-mastercontainer cat /mnt/docker-aio-config/data/configuration.json`, I find the "BORGBACKUP_PASSWORD" key which returns what i am expecting.

I have two AIO installations, and the behaviour is the same.

Did I miss something ?

**Reply by szaimen** [reply]
borg_restore_password is only temporarily used for restoring the instance from backup (and is set to empty again after the restore is done). BORGBACKUP_PASSWORD is what is important to be set.

So everything works as expected.
